packages/synapse-react-client/src/components/SynapseChat/SynapseChat.tsx

```diff
@@ -1,70 +1,122 @@
 import React, { KeyboardEventHandler, useEffect, useState } from 'react'
-import {
-  Box,
-  List,
-  Paper,
-  Typography,
-  FormControl,
-  InputAdornment,
-  IconButton,
-} from '@mui/material'
+import { Box, List, IconButton, Typography } from '@mui/material'
 import { useTheme } from '@mui/material'
 import { ColorPartial } from '@mui/material/styles/createPalette'
-import { OutlinedInput } from '@mui/material'
 import { ArrowUpward } from '@mui/icons-material'
 import SynapseChatInteraction from './SynapseChatInteraction'
 import { SkeletonParagraph } from '../Skeleton'
-import { useCreateAgentSession } from 'src/synapse-queries/chat/useChat'
+import {
+  useCreateAgentSession,
+  useGetAgentChatSessionHistory,
+  useSendChatMessageToAgent,
+} from '../../synapse-queries/chat/useChat'
 import { AgentAccessLevel } from '@sage-bionetworks/synapse-types'
+import { TextField } from '@mui/material'
 
 export type SynapseChatProps = {
-  initialMessage: string
+  initialMessage?: string //optional initial message
   agentId?: string // if provided, use this agent
+  chatbotName?: string // optional name of this chatbot agent
+  hideTitle?: boolean
+  textboxPositionOffset?: string // when embedded in a form, the textbox (form) stuck to the bottom may need to be offset due to container padding (dialog content for example!)
 }
 
+type ChatInteraction = {
+  userMessage: string
+  chatResponseText?: string
+}
 export const SynapseChat: React.FunctionComponent<SynapseChatProps> = ({
   initialMessage,
   agentId,
+  chatbotName = 'SynapseChat',
+  hideTitle = false,
+  textboxPositionOffset = '0px',
 }) => {
-  // Create a new Agent Session on mount, add ability to initialize this component with a starting message from the user.
-  const [agentSessionId, setAgentSessionId] = useState('')
-  const { mutate: createAgentSession } = useCreateAgentSession({
-    onSuccess: newAgentSession => {
-      setAgentSessionId(newAgentSession.sessionId)
+  const { data: agentSession, mutate: createAgentSession } =
+    useCreateAgentSession()
+  const theme = useTheme()
+  // When both the current message and current response are available, add a new ChatInteraction to the array
+  const [interactions, setInteractions] = useState<ChatInteraction[]>([])
+  const [currentInteraction, setCurrentInteraction] =
+    useState<ChatInteraction>()
```

Comment:
I wonder if it might be cleaner to set it up like:

1. Display the historical set of chat interactions & responses
2. When the user submits an interaction, store it in this state variable (rename to `pendingInteraction`) and send the message to the agent.
3. While we have a `pendingInteraction`, show the interaction + a loader for the response
4. Once the response comes back, clear out the `pendingInteraction` state and append the interaction + response to the cached history object (as a sort of optimistic update)

A state variable for `pendingInteraction` might not even be necessary if that state is also tracked and provided by the `useSendChatMessageToAgent` return object.

Thoughts?

Reply:
Definitely renaming currentInteraction to pendingInteraction.  The setup that you describe is what is implemented for steps 1-3, but it does not append the interaction to the cached history object. 
I think I want to keep the server history paginated response separate from the client-side tracked interactions (new interaction data) for now.  This keeps things organized in my head.  In the unsupported case where a session was being interacted with on two (or more) browsers, each browser chat will be isolated and consistent (instead of attempting to keep all in sync with the actual chat history).  But this use case is a terrible reason to keep the existing approach.